src/v2/repositories/implementations/AccountUnificationRepository.ts

```diff
@@ -29,24 +30,37 @@ export class AccountUnificationRepository implements IAccountUnificationReposito
     Guard.ThrowIfUndefined('evmAddress', evmAddress);
 
     const api = await this.api.getApi();
-    // Todo: update the function name once Shibuya runtime has been updated.
     const nativeAddress = api.query.hasOwnProperty('unifiedAccounts')
       ? await api.query.unifiedAccounts.evmToNative<AccountId32>(evmAddress)
-      : '';
+      : toSS58Address(evmAddress);
 
-    return nativeAddress.toString();
+    return nativeAddress.toString() !== '' ? nativeAddress.toString() : toSS58Address(evmAddress);
   }
 
   public async getMappedEvmAddress(nativeAddress: string): Promise<string> {
     Guard.ThrowIfUndefined('nativeAddress', nativeAddress);
 
     const api = await this.api.getApi();
-    // Todo: update the function name once Shibuya runtime has been updated.
     const evmAddress = api.query.hasOwnProperty('unifiedAccounts')
       ? await api.query.unifiedAccounts.nativeToEvm<H160>(nativeAddress)
-      : '';
+      : buildEvmAddress(nativeAddress);
 
-    return evmAddress.toString();
+    return evmAddress.toString() !== '' ? evmAddress.toString() : buildEvmAddress(nativeAddress);
+  }
+
+  public async handleCheckIsUnifiedAccount(address: string): Promise<boolean> {
```

Comment:
This is a general comment for this PR
I see a lot of logic has been added to fetch and check if account is unified. I believe `useAccountUnification` hook can be used with existing logic implemented there
<img width="801" alt="image" src="https://github.com/AstarNetwork/astar-apps/assets/8452361/c08d6776-8f01-4967-9ab1-824adf3e95bc">

When user connects wallet portal already checks if account is unified and stores info in Vuex, so you can reuse data from there for your logic.

Reply:
Thanks for checking. `isAccountUnified` only can check the sender account but we also have to check if the destination account has already been unified when sending ERC20 tokens from EVM to SS58. Therefore, `handleCheckIsUnifiedAccount` function will check it.

Reply:
https://github.com/AstarNetwork/astar-apps/pull/1003#issuecomment-1789169450

Reply:
> Thanks for checking. This is for sending tokens. `isAccountUnified` only can check the sender account but we also have to check if the destination account has already been unified when sending ERC20 tokens from EVM to SS58. Therefore, `handleCheckIsUnifiedAccount` function will check it.

Ok, I understand. For checking receiver is fine. Please use `useAccountUnification` hook for sender where applicable